src/Explorer/Controls/Notebook/NotebookTerminalComponent.tsx

```diff
@@ -2,44 +2,60 @@
  * Wrapper around Notebook server terminal
  */
 
+import postRobot from "post-robot";
 import * as React from "react";
 import * as DataModels from "../../../Contracts/DataModels";
-import * as StringUtils from "../../../Utils/StringUtils";
+import { TerminalProps } from "../../../Terminal/TerminalProps";
 import { userContext } from "../../../UserContext";
-import { TerminalQueryParams } from "../../../Common/Constants";
-import { handleError } from "../../../Common/ErrorHandlingUtils";
+import * as StringUtils from "../../../Utils/StringUtils";
 
 export interface NotebookTerminalComponentProps {
   notebookServerInfo: DataModels.NotebookWorkspaceConnectionInfo;
   databaseAccount: DataModels.DatabaseAccount;
 }
 
 export class NotebookTerminalComponent extends React.Component<NotebookTerminalComponentProps> {
+  private terminalWindow: Window;
+
   constructor(props: NotebookTerminalComponentProps) {
     super(props);
   }
 
+  componentDidMount(): void {
+    this.sendPropsToTerminalFrame();
+  }
+
   public render(): JSX.Element {
     return (
       <div className="notebookTerminalContainer">
-        <iframe
-          title="Terminal to Notebook Server"
-          src={NotebookTerminalComponent.createNotebookAppSrc(this.props.notebookServerInfo, this.getTerminalParams())}
-        />
+        <iframe title="Terminal to Notebook Server" onLoad={this.handleFrameLoad.bind(this)} src="terminal.html" />
```

Comment:
Prob out of scope for this PR, but why is this an `iframe` at all? Can't we load the component async like we do for other notebooks and Monaco? `await import(./notebookTerminal)`

Reply:
@languy probably has more context on this. IMO as long as we lazy load JupyterLab we should be okay not using the `iframe`.